cms/djangoapps/contentstore/views/component.py

```diff
@@ -598,3 +608,70 @@ def component_handler(request, usage_key_string, handler, suffix=''):
         )
 
     return webob_to_django_response(resp)
+
+
+def get_unit_tags(usage_key):
+    """
+    Get the tags of a Unit and build a json to be read by the UI
+    """
+    # Get content tags from content tagging API
+    content_tags = get_content_tags(usage_key)
```

Comment:
@ChrisChV I'm wondering if there is a way to make use of @bradenmacdonald 's updated [implementation](https://github.com/openedx/openedx-learning/pull/111/) in the `openedx-learning` repo that added the lineage information to the object tags and takes care of the grouping under taxonomies as well, rather than re-implementing it here as well we could potentially utilize that as a single source of truth.

Here is a sample of how the data looks like:

```json
{
    "block-v1:SampleTaxonomyOrg1+STC1+2023_1+type@vertical+block@aaf8b8eb86b54281aeeab12499d2cb0b": {
        "taxonomies": [
            {
                "name": "LightCastSkillsTaxonomy",
                "taxonomy_id": 15,
                "editable": true,
                "tags": [
                    {
                        "value": "Administrative Functions",
                        "lineage": [
                            "Administration",
                            "Administrative Support",
                            "Administrative Functions"
                        ]
                    }
                ]
            },
            {
                "name": "OpenCanadaTaxonomy",
                "taxonomy_id": 14,
                "editable": true,
                "tags": [
                    {
                        "value": "Verbal Ability",
                        "lineage": [
                            "Abilities",
                            "Cognitive Abilities",
                            "Communication Abilities",
                            "Verbal Ability"
                        ]
                    }
                ]
            },
            ...
        ]
    }
}
```

Though I think to make use of that, it seems there needs to be some changes made to the `openedx-learning` repo, namely extracting the logic out of the [`ObjectTagsByTaxonomySerializer`](https://github.com/openedx/openedx-learning/blob/7afd44d0e6fd78d2ab0dfa1124f670a7f23c846b/openedx_tagging/core/tagging/rest_api/v1/serializers.py#L93) and exposing a python API that can be used in this repo. Since currently It is only available through the rest api. 

Any thoughts on this?

Reply:
My thinking was that it's OK to have this duplicated here, because eventually this will be re-implemented in the course-authoring MFE (that's out of scope for us now, but at some point ALL of these Studio pages will be using the new MFE only). When that happens, it can use the logic in openedx-learning.

We should definitely put in a comment stating that when this is moved into an MFE it should be simplified to use the REST API which already provides this grouping + sorting logic. But for now since it's already implemented this way, I think it's fine to leave it like this and get it done sooner.

Reply:
Ok @bradenmacdonald, that makes sense :+1: 

@yusuf-musleh Adding another comment, In addition to grouping by taxonomy, it is also necessary to group by parents, to be able to assemble each level of the component.

Reply:
@bradenmacdonald I see, yup, that make sense to me as well!

@ChrisChV Sounds good!